editor/src/clj/editor/animation_set.clj

```diff
@@ -13,118 +13,182 @@
+  {:resource resource
+   :content (protobuf/map->bytes Rig$AnimationSet (:animation-set user-data))})
+
+(g/defnk produce-animation-set-build-target [_node-id resource bones animation-set]
+  (when (not (or (nil? bones)
+                (= 0 (count animation-set))))
+    (bt/with-content-hash
+       {:node-id _node-id
+        :resource (workspace/make-build-resource resource)
+        :build-fn build-animation-set
+        :user-data {:animation-set animation-set}})))
 
 (def ^:private form-sections
   {:navigation false
    :sections
    [{:title "Animation Set"
-     :fields [{:path [:animations]
+     :fields [{:path [:skeleton]
+               :type :resource
+               :filter model-scene/model-file-types
+               :label "Skeleton"
+               :default nil}
+              {:path [:animations]
                :type :list
                :label "Animations"
                :element {:type :resource
-                         :filter #{"animationset" "dae"}
+                         :filter model-scene/animation-file-types
                          :default nil}}]}]})
 
-(defn- set-form-op [{:keys [node-id]} path value]
-  (assert (= path [:animations]))
-  (g/set-property! node-id :animations value))
+(defn- set-form-op [{:keys [node-id]} [property] value]
+  (g/set-property! node-id property value))
 
-(g/defnk produce-form-data [_node-id animations]
-  (let [values {[:animations] animations}]
+(defn- clear-form-op [{:keys [node-id]} [property]]
+  (g/clear-property! node-id property))
+
+(g/defnk produce-form-data [_node-id skeleton animation-resources]
+  (let [values {[:skeleton] skeleton
+                [:animations] animation-resources}]
     (-> form-sections
         (assoc :form-ops {:user-data {:node-id _node-id}
-                          :set set-form-op})
+                          :set set-form-op
+                          :clear clear-form-op})
         (assoc :values values))))
 
 (g/defnode AnimationSetNode
   (inherits resource-node/ResourceNode)
 
+  (input skeleton-resource resource/Resource)
+  (input animations-resource resource/Resource)
   (input animation-resources resource/Resource :array)
   (input animation-sets g/Any :array)
+  (input animation-infos g/Any :array)
+  (output animation-info g/Any :cached produce-animation-info)
+           
+  (input bones g/Any)
+  (output bones g/Any (gu/passthrough bones))
+
+  (property skeleton resource/Resource
+            (value (gu/passthrough skeleton-resource))
+            (set (fn [evaluation-context self old-value new-value]
+                   (project/resource-setter evaluation-context self old-value new-value
+                                            [:resource :skeleton-resource]
+                                            [:bones :bones])))
+            (dynamic error (g/fnk [_node-id skeleton]
+                                  (or (validation/prop-error :info _node-id :skeleton validation/prop-nil? skeleton "Skeleton")
```

Comment:
Currently, this doesn't seem to generate a build error when the skeleton is missing. Not sure why 🤔